pkg/machines/libvirt-dbus.es6

```diff
@@ -0,0 +1,861 @@
+                        if (!updatedXml) {
+                            dispatch(vmActionFailed({
+                                name,
+                                connectionName,
+                                message: _("VM CHANGE_NETWORK_STATE action failed: updated device XML couldn't not be generated"),
+                            }));
+                        } else {
+                            clientLibvirt[connectionName].call(objPath, 'org.libvirt.Domain', 'UpdateDevice', [updatedXml, 0], TIMEOUT)
+                                    .done(() => {
+                                        dispatch(getVm({connectionName, id:objPath}));
+                                    })
+                                    .fail(exception => dispatch(vmActionFailed({
+                                        name,
+                                        connectionName,
+                                        message: _("VM CHANGE_NETWORK_STATE action failed"),
+                                        detail: {exception}
+                                    })));
+                        }
+                    });
+        };
+    },
+
+    CREATE_AND_ATTACH_VOLUME({
+        connectionName,
+        poolName,
+        volumeName,
+        size,
+        format,
+        target,
+        vmId,
+        permanent,
+        hotplug
+    }) {
+        let volXmlDesc = getVolumeXML(volumeName, size, format, target);
+
+        return dispatch => {
+            clientLibvirt[connectionName].call('/org/libvirt/QEMU', 'org.libvirt.Connect', 'StoragePoolLookupByName', [poolName], TIMEOUT)
+                    .done((storagePoolPath) => {
+                        clientLibvirt[connectionName].call(storagePoolPath[0], 'org.libvirt.StoragePool', 'StorageVolCreateXML', [volXmlDesc, 0], TIMEOUT)
+                                .done((storageVolumePath) => {
+                                    clientLibvirt[connectionName].call(storageVolumePath[0], "org.freedesktop.DBus.Properties", "GetAll", ["org.libvirt.StorageVol"], TIMEOUT)
+                                            .done(function(props) {
+                                                if (!$.isEmptyObject(props[0]))
+                                                    return dispatch(attachDisk({ connectionName, diskFileName: props[0].Path.v.v, target, vmId, permanent, hotplug }))
+                                                            .then(() => {
+                                                                // force reload of VM data, events are not reliable (i.e. for a down VM)
+                                                                dispatch(getVm({connectionName, id:vmId}));
+                                                            });
+                                            })
+                                            .fail(function(ex) { console.warn("failed waiting fot the StorageVol proxy to get ready", ex) });
```

Comment:
All these console warns and failures should be replaced with exceptions visible to UI -> dispatch(vmActionFailed...

Reply:
Done.
However, now UI error appear in the Vm Overview subtab. Ideally it should be viewed on the subtab where they where actually caused.
NOTE: libvirt-virsh implementation doesn't show any exceptions to user from attach-disk and create-and-attach-disk